src/modals/storage-connections/StorageConnectionsDataProvider.js

```diff
@@ -0,0 +1,119 @@
+import { engineGet, enginePost, enginePut, engineDelete } from '_/utils/fetch'
+import DataProvider from '_/components/helper/DataProvider'
+import { webadminToastTypes } from '_/constants'
+import { msg } from '_/intl-messages'
+
+const createConnection = async (connection) => {
+  return await enginePost(
+    'api/storageconnections',
+    JSON.stringify(connection)
+  )
+}
+
+const editConnection = async (connection, connectionId) => {
+  return await enginePut(
+    `api/storageconnections/${connectionId}`,
+    JSON.stringify(connection)
+  )
+}
+
+const deleteConnection = async (connectionId) => {
+  return await engineDelete(`api/storageconnections/${connectionId}`)
+}
+
+const attachConnection = async (connectionId, domainId) => {
+  return await enginePost(
+    `api/storagedomains/${domainId}/storageconnections`,
+    JSON.stringify({ id: connectionId })
+  )
+}
+
+const detachConnection = async (connectionId, domainId) => {
+  return await engineDelete(`api/storagedomains/${domainId}/storageconnections/${connectionId}`)
+}
+
+const fetchData = async (storageDomainId) => {
+  const allConnectionsJson = await engineGet('api/storageconnections')
+  let storageDomain = await engineGet(`api/storagedomains/${storageDomainId}`)
+  const domainConnectionsJson = await engineGet(`api/storagedomains/${storageDomainId}/storageconnections`)
+
+  const dataCenter = storageDomain.data_centers.data_center[0]
+  if (!dataCenter) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch data center')
+  }
+
+  storageDomain = await engineGet(`api/datacenters/${dataCenter.id}/storagedomains/${storageDomainId}`)
+
+  if (!storageDomain) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch storage domain')
+  }
+
+  let allConnections = allConnectionsJson.storage_connection
+  let domainConnections = domainConnectionsJson.storage_connection
+
+  if (!allConnections || allConnections.error) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch storage connections')
+  }
+
+  if (!domainConnections || domainConnections.error) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch storage connections' +
+      'for storage domain ' + storageDomainId)
+  }
+
+  const domainConnectionsIds = new Set(domainConnections.map(connection => connection.id))
+  const isConnectionAttachedToDomain = (conn) => { return domainConnectionsIds.has(conn.id) }
+  const isConnectionNotAttachedToDomain = (conn) => { return !isConnectionAttachedToDomain(conn) }
+  const allConnectionsByType = allConnections
+    .filter(connection => connection.type === storageDomain.storage.type)
+    .map((conn) => {
+      conn.isAttachedToDomain = isConnectionAttachedToDomain(conn)
+      return conn
+    })
+  domainConnections = allConnectionsByType.filter(isConnectionAttachedToDomain)
+  allConnections = domainConnections.concat(allConnectionsByType.filter(isConnectionNotAttachedToDomain))
```

Comment:
it seems to me that at this point `allConnection` should be simply `allConectionByType`